src/Plugin/RedirectPlugin.php

```diff
@@ -230,36 +230,18 @@ private function createUri(ResponseInterface $redirectResponse, RequestInterface
             throw new HttpException(sprintf('Location %s could not be parsed', $location), $originalRequest, $redirectResponse);
         }
 
-        $uri = $originalRequest->getUri();
-
-        if (array_key_exists('scheme', $parsedLocation)) {
-            $uri = $uri->withScheme($parsedLocation['scheme']);
-        }
+        $uri = $originalRequest->getUri()
+            ->withPath(array_key_exists('path', $parsedLocation) ? $parsedLocation['path'] : '')
+            ->withQuery(array_key_exists('query', $parsedLocation) ? $parsedLocation['query'] : '')
+            ->withFragment(array_key_exists('fragment', $parsedLocation) ? $parsedLocation['fragment'] : '')
+            ->withScheme(array_key_exists('scheme', $parsedLocation) ? $parsedLocation['scheme'] : 'http')
+            ->withPort(array_key_exists('port', $parsedLocation) ? $parsedLocation['port'] : null)
```

Comment:
i think scheme and port can not be handled like this. if there was no domain in the redirect at all, we should keep scheme and port from the original domain.

Reply:
I thought you agreed on this when you commented this
> i am actually not sure about this one either. if i was on http://localhost:8080/foo and i get redirect a header [php-http.org/bar](https://php-http.org/bar) i would not expect to stay on port 8080... the only case where the port should be kept is when there is also no host in the redirection... wdyt?

Reply:
I think taking over scheme/port will cause same class of bug for somebody else as I had with path. If hosts change, HTTP client should not use same mechanism as browser does when hosts don't change.

Reply:
yes, we agreed on that. but there are more cases:
* `Location http://otherhost/foobar` => drop the port.
* `Location ://otherhost/foobar` => drop port, keep scheme
* `Location `/foobar` => do not touch scheme, host, port

Reply:
fair enough. sorry for the feature creep. i created https://github.com/php-http/client-common/issues/219